File: S2_KP/lab3/Figure.cpp
#include <iostream>
#include "declarations.hpp"



int main (int argc, char *argv[]) {
    std::string figures = convertToString(argv[1]);
    int l = figures.length();
    int position = 2;

    for (int i = 0; i < l; i++) {
        if (figures[i] == 'c') { // circle
            double r = std::stod(convertToString(argv[position++]));
            Figure *c = new Circle(r);
            c->PrintValues(i + 1,"circle",c->Length(),c->Surface());
            delete c;
        }
        else if (figures[i] == 'q') { // quadrangle
            double a = std::stod(convertToString(argv[position]));
            double phi = std::stod(convertToString(argv[position + 4]));
            if (phi == 90) {
                double b = std::stod(convertToString(argv[position + 1]));
                if (a != b) { // rectangle
                    Figure *r = new Rectangle(a,b);
                    r->PrintValues(i + 1,"rectangle",r->Length(),r->Surface());
                    delete r;
                }
                else {
                    b = std::stod(convertToString(argv[position + 2]));
                    if (a != b) { // rectangle
                        Figure *r = new Rectangle(a,b);
                        r->PrintValues(i + 1,"rectangle",r->Length(),r->Surface());
                        delete r;
                    }
                    else { // square
                        Figure *s = new Square(a);
                        s->PrintValues(i + 1,"square",s->Length(),s->Surface());
                        delete s;
                    }
                }        
            }
            else { // rhomb
                Figure *rh = new Rhomb(a,phi);
                rh->PrintValues(i + 1,"rhomb",rh->Length(),rh->Surface());
                delete rh;
            }
            position += 5;
        }
        else if (figures[i] == 'p') { // pentagon
            double a = std::stod(convertToString(argv[position++]));
            Figure *p = new Pentagon(a);
            p->PrintValues(i + 1,"pentagon",p->Length(),p->Surface());
            delete p;
        }
        else if (figures[i] == 'h') { // hexagon
            double a = std::stod(convertToString(argv[position++]));
            Figure *h = new Hexagon(a);
            h->PrintValues(i + 1,"hexagon",h->Length(),h->Surface());
            delete h;
        }
        else {
            std::cout << "Niepoprawne dane" << std::endl;
            break;
        }
    }

    return 0;
}

File: S2_KP/lab3/Figure.java
interface FigureValues {
    double Length();
    double Surface();
}



public abstract class Figure implements FigureValues {
    public void PrintValues (int nr, String figure, double L, double S) {
        System.out.println("Object " + nr + ": " + figure);
        System.out.println("\tLength = " + L);
        System.out.println("\tSurface = " + S);      
    }
}



abstract class Quadrangle extends Figure {}



class Square extends Quadrangle {
    private double a;
    Square (double a) {
        this.a = a;
    }
    @Override 
    public double Length() {
        return 4 * this.a;
    }
    @Override 
    public double Surface() {
        return this.a * this.a;
    }
}



class Rectangle extends Quadrangle {
    private double a, b;
    Rectangle (double a, double b) {
        this.a = a;
        this.b = b;
    }
    @Override 
    public double Length() {
        return 2 * (this.a + this.b);
    }
    @Override 
    public double Surface() {
        return this.a * this.b;
    }
} 



class Rhomb extends Quadrangle {
    private double a, phi;
    Rhomb (double a, double phi) {
        this.a = a;
        this.phi = phi;
    }
    @Override 
    public double Length() {
        return 4 * this.a;
    }
    @Override 
    public double Surface() {
        return this.a * this.a * Math.sin(phi * Math.PI / 180);
    }
}



class Circle extends Figure {
    private double r;
    Circle (double r) {
        this.r = r;
    }
    @Override 
    public double Length() {
        return 2 * Math.PI * this.r;
    }
    @Override 
    public double Surface() {
        return Math.PI * this.r * this.r;
    }
}



class Pentagon extends Figure {
    private double a;
    Pentagon (double a) {
        this.a = a;
    }
    @Override 
    public double Length() {
        return 5 * this.a;
    }
    @Override 
    public double Surface() {
        return 5 * this.a * this.a * Math.sqrt(3) / 4;
    }
}



class Hexagon extends Figure {
    private double a;
    Hexagon (double a) {
        this.a = a;
    }
    @Override 
    public double Length() {
        return 6 * this.a;
    }
    @Override 
    public double Surface() {
        return 3 * this.a * this.a * Math.sqrt(3) / 2;
    }
}